client/asset/zec/zec.go

```diff
@@ -285,14 +294,207 @@ func connect(c rpcCaller) error {
+	return w.sendOne(ctx, anyTAddr, oneAddr, transparentVal, AllowRevealedSenders)
+}
+
+// UnshieldFunds moves funds from the shielded account to the transparent
+// account.
+func (w *zecWallet) UnshieldFunds(ctx context.Context, amt uint64) (dex.Bytes, error) {
+	// lastAddr, err := w.lastShieldedAddress()
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	bal, err := zGetBalanceForAccount(w, shieldedAcctNumber)
+	if err != nil {
+		return nil, fmt.Errorf("z_getbalance error: %w", err)
+	}
+	const fees = 1000
+	if bal < fees || bal-fees < amt {
+		return nil, asset.ErrInsufficientBalance
+	}
+
+	unified, err := zGetAddressForAccount(w, transparentAcctNumber, []string{transparentAddressType, orchardAddressType})
+	if err != nil {
+		return nil, fmt.Errorf("z_getaddressforaccount error: %w", err)
+	}
+
+	receivers, err := zGetUnifiedReceivers(w, unified.Address)
+	if err != nil {
+		return nil, fmt.Errorf("z_getunifiedreceivers error: %w", err)
+	}
+
+	return w.sendOneShielded(ctx, receivers.Transparent, amt, AllowRevealedRecipients)
+}
+
+// sendOne is a helper function for doing a z_sendmany with a single recipient.
+func (w *zecWallet) sendOne(ctx context.Context, fromAddr, toAddr string, amt uint64, priv privacyPolicy) (dex.Bytes, error) {
+	recip := singleSendManyRecipient(toAddr, amt)
+
+	operationID, err := zSendMany(w, fromAddr, recip, priv)
+	if err != nil {
+		return nil, fmt.Errorf("z_sendmany error: %w", err)
+	}
+
+	txid, err := awaitSendManyOperation(ctx, w, operationID)
+	if err != nil {
+		return nil, err
+	}
+
+	return txid[:], nil
+}
+
+func (w *zecWallet) sendOneShielded(ctx context.Context, toAddr string, amt uint64, priv privacyPolicy) (dex.Bytes, error) {
+	lastAddr, err := w.lastShieldedAddress()
+	if err != nil {
+		return nil, err
+	}
+	return w.sendOne(ctx, lastAddr, toAddr, amt, priv)
+}
+
+// SendShielded sends funds from the shielded account to the provided shielded
+// or transparent address.
+func (w *zecWallet) SendShielded(ctx context.Context, toAddr string, amt uint64) (dex.Bytes, error) {
+	bal, err := zGetBalanceForAccount(w, shieldedAcctNumber)
+	if err != nil {
+		return nil, err
+	}
+	const fees = 1000
+	if bal < fees || bal-fees < amt {
+		return nil, asset.ErrInsufficientBalance
+	}
+
+	res, err := zValidateAddress(w, toAddr)
+	if err != nil {
+		return nil, fmt.Errorf("error validating address: %w", err)
+	}
+
+	var priv privacyPolicy
+	if res.AddressType == unifiedAddressType {
+		// Could be orchard, or other unified..
+		receivers, err := zGetUnifiedReceivers(w, toAddr)
+		if err != nil {
+			return nil, fmt.Errorf("error getting unified receivers: %w", err)
+		}
+		toAddr, priv, err = receivers.favoredAddressAndPolicy()
+		if err != nil {
+			return nil, fmt.Errorf("error parsing unified receiver: %w", err)
+		}
+	} else if res.AddressType == transparentAddressType {
+		priv = AllowRevealedRecipients
+	} else if res.AddressType == saplingAddressType {
+		priv = AllowRevealedAmounts
```

Comment:
These two policies look right, so I guess they are switched in `receivers.favoredAddressAndPolicy`